### PR TITLE
feat: validate proposal threshold when creating a proposal

### PR DESCRIPTION
**Pull request by Sekhmet**
## Summary

Closes: https://github.com/snapshot-labs/sx-ui/issues/460

This PR adds check to proposal threshold on proposal editor.

## Test plan
- Try creating proposal here http://localhost:8080/#/gor:0xd9e572dfe8f188dc34dbc5adc5413ab695b684c5
- It should show error message for all accounts.
- Try creating proposal here http://localhost:8080/#/gor:0x71dc8ba1121d00e48d8fca1558895205845f1463
- It shouldn't show error

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
   content: ['./index.html', './src/**/*.{js,ts,vue}'],
+  darkMode: 'class',
```
Comment (Sekhmet) [inline tailwind.config.js:3]: Needed so we can use `dark:` classes.

File: src/views/Editor.vue [hunk 2]
```diff
@@ -49,6 +99,8 @@ onMounted(() => {
   }
 });
 
+watch([space, () => web3.value.account], () => getVotingPower());
```
Comment (j0hnfl0w) [inline src/views/Editor.vue:102]: Nice one-liner 🙂

**Comment by bonustrack** [comment]
On http://localhost:8080/#/gor:0x71dc8ba1121d00e48d8fca1558895205845f1463 it show an error for me
image

File: src/views/Editor.vue [hunk 3]
```diff
@@ -114,6 +147,9 @@ async function handleProposeClick() {
       </div>
     </nav>
     <Container v-if="proposals[proposalKey]" class="pt-5 s-box">
+      <UiAlert v-if="!fetchingVotingPower && !votingPowerValid" type="error">
+        You do not have enough Voting Power to create proposal in this space.
```
Comment (bonustrack) [inline src/views/Editor.vue:151]: We dont need the uppercase here for voting power
```suggestion
        You do not have enough voting power to create proposal in this space.
```

File: src/views/Editor.vue [hunk 4]
```diff
@@ -34,6 +41,49 @@ if (!proposals[proposalKey]) {
...
+      proposals[`${id}:${key}`].execution
+    );
+
+    if (result) router.back();
+  } finally {
+    sending.value = false;
+  }
+}
+
+async function getVotingPower() {
```
Comment (bonustrack) [inline src/views/Editor.vue:64]: We may want to reduce logic on view pages and move that to helpers

File: src/views/Editor.vue [hunk 5]
```diff
@@ -34,6 +41,49 @@ if (!proposals[proposalKey]) {
...
+
+  sending.value = true;
+
+  try {
+    const result = await propose(
+      space.value,
+      proposals[`${id}:${key}`].title,
+      proposals[`${id}:${key}`].body,
+      proposals[`${id}:${key}`].discussion,
+      proposals[`${id}:${key}`].execution
```
[the remark this answers is not in the document]
Comment (j0hnfl0w) [reply src/views/Editor.vue:55]: we even have one above ```const proposalKey = `${id}:${key}`;```

File: src/views/Editor.vue [hunk 6]
```diff
@@ -104,7 +136,8 @@ async function handleProposeClick() {
           </UiButton>
           <UiButton
             class="rounded-l-none border-l-0 float-left !m-0 !px-3"
-            :loading="sending"
+            :loading="sending || (web3.account !== '' && fetchingVotingPower)"
```
[the remark this answers is not in the document]
Comment (Sekhmet) [reply src/views/Editor.vue:138]: We don't have other choice, it defaults to `''` :D

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**